internal/receiver/scriptedinputsreceiver/procpipe/procpipe.go

```diff
@@ -0,0 +1,233 @@
+// limitations under the License.
+
+package procpipe
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"go.uber.org/zap"
+)
+
+const (
+	operatorType = "procpipe_input"
+
+	// minMaxLogSize is the minimal size which can be used for buffering
+	// TCP input
+	minMaxLogSize = 64 * 1024
+
+	// DefaultMaxLogSize is the max buffer sized used
+	// if MaxLogSize is not set
+	DefaultMaxLogSize = 1024 * 1024
+
+	DefaultIntervalSeconds = 60
+)
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
+}
+
+// Build will build a stdin input operator.
+func (c *Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(logger)
+	if err != nil {
+		return nil, err
+	}
+
+	// If MaxLogSize not set, set sane default
+	if c.MaxLogSize == 0 {
+		c.MaxLogSize = DefaultMaxLogSize
+	}
+
+	if c.CollectionInterval == "" {
+		c.CollectionInterval = strconv.Itoa(DefaultIntervalSeconds)
+	}
+
+	encoding, err := c.Encoding.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	// Build multiline
+	var splitFunc bufio.SplitFunc
+	if c.Multiline.LineStartPattern == "" && c.Multiline.LineEndPattern == "" {
+		splitFunc = helper.SplitNone(int(c.MaxLogSize))
+	} else {
+		splitFunc, err = c.Multiline.Build(encoding.Encoding, true, false, false, nil, int(c.MaxLogSize))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &Input{
+		baseConfig:    c.BaseConfig,
+		InputOperator: inputOperator,
+		logger:        logger,
+		MaxLogSize:    int(c.MaxLogSize),
+		encoding:      encoding,
+		splitFunc:     splitFunc,
+	}, nil
+}
+
+// Input is an operator that reads input from stdin
+type Input struct {
+	logger    *zap.SugaredLogger
+	cancelAll context.CancelFunc
+	splitFunc bufio.SplitFunc
+	helper.InputOperator
+	encoding   helper.Encoding
+	baseConfig BaseConfig
+	wg         sync.WaitGroup
+	MaxLogSize int
+}
+
+func createTicker(intervalStr string) (*time.Ticker, error) {
+	interval, err := time.ParseDuration(intervalStr)
+	if err != nil {
+		return nil, err
+	}
+	return time.NewTicker(interval), nil
+}
+
+// Start will start generating log entries.
+func (g *Input) Start(_ operator.Persister) error {
+
+	ctx, cancelAll := context.WithCancel(context.Background())
+	g.cancelAll = cancelAll
+
+	ticker, err := createTicker(g.baseConfig.CollectionInterval)
+	if err != nil {
+		// TODO: move ticker verification to Start() so that we cannot error here.
+		return err
+	}
+
+	go func() {
+		for {
+			_, cancelCycle := context.WithCancel(ctx)
+
+			err := g.beginCycle(ctx)
+			if err != nil {
+				g.logger.Errorf("Error running script: %v", err)
+			}
+
+			select {
+			case <-ctx.Done():
+				cancelCycle()
+				return
+			case <-ticker.C:
+				cancelCycle()
+				g.wg.Wait()
```

Comment:
So if I use scraper helper I could totally replace custom main loop and termination logic present in procpipe? If I use scraper helper would I still need to introduce sentinel channel and poison pill?

Reply:
Sorry, misread this. I think the scraper helper provides most of the functionality and you'd just need to define a scrape method.

Reply:
Apologies, this should have been obvious. refining the existing commander seems like the reasonable path forward, though I think we may want to separate its stderr handling for status assessments instead of forwarding all for operator processing.